### PR TITLE
Add intent for switching from places to map view and corresponding test

### PR DESCRIPTION
**Pull request by vforteli**
Connected to #161

### CODE REVIEW

**Comment by vforteli** [comment]
Brilliant, passed locally but not on CI...

File: App/src/main/java/cc/softwarefactory/lokki/android/fragments/PlacesFragment.java [hunk 1]
```diff
@@ -223,7 +224,7 @@ public void onClick(View v) {
                 MainApplication.emailBeingTracked = email;
                 Intent intentLocation = new Intent("LOCATION-UPDATE");
                 LocalBroadcastManager.getInstance(getActivity()).sendBroadcast(intentLocation);
-                Intent intentMapTab = new Intent("GO_TO_MAP_TAB");
+                Intent intentMapTab = new Intent("GO_TO_MAP");
```
Comment (Kauhsa) [inline App/src/main/java/cc/softwarefactory/lokki/android/fragments/PlacesFragment.java]: Variable name is old :]

**Comment by Kauhsa** [comment]
Hmmh. The menu stays at places, header text is places and invisibility button is missing even when we're at map screen. So I guess we don't *really* go to the map screen? Do we want this? Personally, I think having a "kinda in map screen but not quite" state in the app is just confusing.

**Comment by vforteli** [comment]
I made a separate issue about that. It has apparently always worked like
this

File: App/src/test/java/cc/softwarefactory/lokki/android/espresso/PlacesScreenTest.java [hunk 2]
```diff
@@ -74,6 +84,23 @@ public void testContactAppearsInPlace() throws JSONException {
     }
 
 
+    public void testClickContactOpensMap() throws JSONException, InterruptedException {
+        getMockDispatcher().setPlacesResponse(new MockResponse().setBody(MockJsonUtils.getPlacesJson()));
+        String[] contactEmails = (new String[]{"family.member@example.com"});
```
Comment (Kauhsa) [inline App/src/test/java/cc/softwarefactory/lokki/android/espresso/PlacesScreenTest.java]: I don't think you need braces here

File: App/src/test/java/cc/softwarefactory/lokki/android/espresso/PlacesScreenTest.java [hunk 3]
```diff
@@ -74,6 +84,23 @@ public void testContactAppearsInPlace() throws JSONException {
...
+        JSONObject location = new JSONObject();
+        location.put("lat", "37.483477313364574") //Testplace1
+                .put("lon", "-122.14838393032551")
+                .put("rad", "100");
+        JSONObject[] locations = (new JSONObject[]{location});
```
Comment (Kauhsa) [inline App/src/test/java/cc/softwarefactory/lokki/android/espresso/PlacesScreenTest.java]: Here neither

File: App/src/test/java/cc/softwarefactory/lokki/android/espresso/PlacesScreenTest.java [hunk 4]
```diff
@@ -74,6 +84,23 @@ public void testContactAppearsInPlace() throws JSONException {
     }
 
 
+    public void testClickContactOpensMap() throws JSONException, InterruptedException {
+        getMockDispatcher().setPlacesResponse(new MockResponse().setBody(MockJsonUtils.getPlacesJson()));
+        String[] contactEmails = (new String[]{"family.member@example.com"});
+        JSONObject location = new JSONObject();
+        location.put("lat", "37.483477313364574") //Testplace1
```
Comment (Kauhsa) [inline App/src/test/java/cc/softwarefactory/lokki/android/espresso/PlacesScreenTest.java]: What's this comment? Do we use these coordinates elsewhere as well? We probably want to put them into constant in that case.

**Comment by Kauhsa** [comment]
Made some minor comments, but other than that, it seems to work now fine. The index thing going on with the menu is pretty insane, though, but outside of this PR.

**Comment by Kauhsa** [comment]
Apparently this waits for #177?

File: App/src/main/java/cc/softwarefactory/lokki/android/activities/MainActivity.java [hunk 5]
```diff
@@ -131,6 +131,19 @@ protected void onResume() {
         Log.e(TAG, "onResume - NOT firstTimeLaunch, so launching services.");
         startServices();
         LocalBroadcastManager.getInstance(this).registerReceiver(exitMessageReceiver, new IntentFilter("EXIT"));
+        LocalBroadcastManager.getInstance(this).registerReceiver(switchToMapReceiver, new IntentFilter("GO_TO_MAP"));
+    }
+
+
+
+
```
Comment (josalmi) [inline App/src/main/java/cc/softwarefactory/lokki/android/activities/MainActivity.java]: Too many new lines

File: App/src/main/java/cc/softwarefactory/lokki/android/fragments/PlacesFragment.java [hunk 6]
```diff
@@ -146,6 +146,7 @@ public boolean onLongClick(View v) {
                                 Log.e(TAG, "Avatar not in cache, email: " + email);
                                 image.setImageResource(R.drawable.default_avatar);
                             }
+                            image.setContentDescription(email); // This is here to easier test the feature. Also accessibility guidelines recommend using this for uniquely identifying items
```
Comment (josalmi) [inline App/src/main/java/cc/softwarefactory/lokki/android/fragments/PlacesFragment.java]: You could remove the comment

File: App/src/main/java/cc/softwarefactory/lokki/android/fragments/PlacesFragment.java [hunk 7]
```diff
@@ -221,10 +222,8 @@ private RoundedImageView createAvatar(final String email) {
             public void onClick(View v) {
 
                 MainApplication.emailBeingTracked = email;
-                Intent intentLocation = new Intent("LOCATION-UPDATE");
-                LocalBroadcastManager.getInstance(getActivity()).sendBroadcast(intentLocation);
-                Intent intentMapTab = new Intent("GO_TO_MAP_TAB");
-                LocalBroadcastManager.getInstance(getActivity()).sendBroadcast(intentMapTab);
+                LocalBroadcastManager.getInstance(getActivity()).sendBroadcast(new Intent("LOCATION-UPDATE"));
+                LocalBroadcastManager.getInstance(getActivity()).sendBroadcast(new Intent("GO_TO_MAP"));
```
Comment (josalmi) [inline App/src/main/java/cc/softwarefactory/lokki/android/fragments/PlacesFragment.java]: - and _ mixed. No likey

File: App/src/test/java/cc/softwarefactory/lokki/android/espresso/PlacesScreenTest.java [hunk 8]
```diff
@@ -74,6 +84,23 @@ public void testContactAppearsInPlace() throws JSONException {
...
+        JSONObject[] locations = new JSONObject[]{location};
+        getMockDispatcher().setDashboardResponse(new MockResponse().setBody(MockJsonUtils
+                .getDashboardJsonContactsUserLocation(contactEmails, locations, location)));
+        enterPlacesScreen();
+        onView(withContentDescription("family.member@example.com")).perform(click());
+
+        onView(withId(R.id.map)).check(matches(isDisplayed()));
+    }
+
+
```
Comment (josalmi) [inline App/src/test/java/cc/softwarefactory/lokki/android/espresso/PlacesScreenTest.java]: One new line is probably enough

**Pull request merged**